### PR TITLE
Properly type enums for @types/steam-tradeoffer-manager

### PR DESCRIPTION
**Pull request by Nicklason**
The enums could be done similarly to other packages, like steam-user and steamcommunity but they don't have the mapping of enum value to key which can't be done with the enum type.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5f3288c3162af0aa229d661540dd6dd87478fd91/types/steam-user/index.d.ts#L1449-L1458

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5f3288c3162af0aa229d661540dd6dd87478fd91/types/steamcommunity/index.d.ts#L710-L721

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DoctorMcKay/node-steam-tradeoffer-manager/tree/master/resources

### CODE REVIEW

File: types/steam-tradeoffer-manager/steam-tradeoffer-manager-tests.ts [hunk 1]
```diff
@@ -13,6 +13,23 @@ const steam = new Steam.SteamClient();
 const offer = manager.createOffer("123");
 const item = new CEconItem("a", "b", "c");
 
+// ----- Enums -----
```
Comment (kjsmita6) [inline types/steam-tradeoffer-manager/steam-tradeoffer-manager-tests.ts:16]: Add test cases for the number keys as well.

```js
ETradeOfferState[2] == 'Active'
```
etc

**Review by kjsmita6 — CHANGES_REQUESTED** [review]

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/steam-tradeoffer-manager/steam-tradeoffer-manager-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f57f5927214bdb2be384b1ed32aefe39c324860b
</p>

**Review by kjsmita6 — APPROVED** [review]
Sorry for the long delay, been busy coding at my work irl :)

Looks good, thanks again!

**Comment by Nicklason** [comment]
Ready to merge

**Pull request merged**